jobbergate-agent/tests/jobbergate/test_compute.py

```diff
@@ -0,0 +1,307 @@
+"""Define tests for the functions in jobbergate_agent/utils/compute.py."""
+
+import pytest
+import random
+import time
+from collections.abc import Callable
+from datetime import datetime
+from typing import cast, get_args
+from unittest import mock
+
+import numpy as np
+
+from jobbergate_agent.jobbergate.constants import INFLUXDB_MEASUREMENT
+from jobbergate_agent.jobbergate.schemas import InfluxDBMeasure
+from jobbergate_agent.utils.compute import (
+    aggregate_influx_measures,
+    measure_memory_usage,
+    _create_mapping,
+    _aggregate_with_numba,
+)
+
+
+@pytest.fixture()
+def generate_and_aggregate_job_metrics_data() -> (
+    Callable[
+        [int, int, int, int, int],
+        tuple[
+            list[InfluxDBMeasure],
+            list[tuple[int, str, str, str, float, float, float, float, float, float, float, float, float, float]],
+        ],
+    ]
+):
+    """
+    Generates sample InfluxDB data and its aggregated counterpart.
+
+    Returns a function that creates both the list of measures and their aggregated version.
+    """
+
+    def _generate_and_aggregate(
+        num_points_per_measurement: int, num_hosts: int, num_jobs: int, num_steps: int, num_tasks: int
+    ) -> tuple[
+        list[InfluxDBMeasure],
+        list[tuple[int, str, str, str, float, float, float, float, float, float, float, float, float, float]],
+    ]:
+        # Initialize data structures
+        current_time = int(datetime.now().timestamp())
+        measurement_names = get_args(INFLUXDB_MEASUREMENT)
+        default_measurements: dict[str, float] = {measurement: 0.0 for measurement in measurement_names}
+
+        measures = []
+        aggregated_data: dict[tuple[int, str, str, str], dict[str, float]] = {}
+
+        # Generate measures
+        for _ in range(num_points_per_measurement):
+            for host in range(1, num_hosts + 1):
+                for job in range(1, num_jobs + 1):
+                    for step in range(1, num_steps + 1):
+                        for task in range(1, num_tasks + 1):
+                            key = (current_time, f"host_{host}", str(step), str(task))
+
+                            if key not in aggregated_data:
+                                aggregated_data[key] = default_measurements.copy()
+
+                            for measurement in measurement_names:
+                                value = random.random() * 100
+                                measure = InfluxDBMeasure(
```

Comment:
If this were a dataclass, you could use keyword arguments here 😉

Reply:
we should have some standard for this cases, otherwise it just falls in personal preference. I don't mind the way it is honestly